src/global/MethodTest.java

```diff
@@ -0,0 +1,241 @@
+        this.methodClass = methodClass;
+        this.methodName = methodName;
+        this.arguments = arguments;
+        this.methodArgumentTypes = methodArgumentTypes;
+        this.methodTestSentence = methodTestSentence;
+        this.incorrectMethodStructureErrorMessage = "Your " + methodName + " method's output did not follow the correct structure/syntax";
+        setMethodOutput();
+    }
+
+    public MethodTest(Class<?> methodClass, String methodName, Clause[] methodTestSentence) {
+        this.methodClass = methodClass;
+        this.methodName = methodName;
+        this.arguments = null;
+        this.methodArgumentTypes = null;
+        this.methodTestSentence = methodTestSentence;
+        this.incorrectMethodStructureErrorMessage = "Your " + methodName + " method's output did not follow the correct structure/syntax";
+        setMethodOutput();
+    }
+
+    public MethodTest(Class<?> methodClass, String methodName, Object[] arguments, Class<?>[] methodArgumentTypes, String incorrectMethodStructureErrorMessage) {
+        this.methodClass = methodClass;
+        this.methodName = methodName;
+        this.arguments = arguments;
+        this.methodArgumentTypes = methodArgumentTypes;
+        this.methodTestSentence = null;
+        this.incorrectMethodStructureErrorMessage = incorrectMethodStructureErrorMessage;
+        setMethodOutput();
+    }
+
+    public MethodTest(Class<?> methodClass, String methodName, String incorrectMethodStructureErrorMessage) {
+        this.methodClass = methodClass;
+        this.methodName = methodName;
+        this.arguments = null;
+        this.methodArgumentTypes = null;
+        this.methodTestSentence = null;
+        this.incorrectMethodStructureErrorMessage = incorrectMethodStructureErrorMessage;
+        setMethodOutput();
+    }
+
+    public MethodTest(Class<?> methodClass, String methodName, Object[] arguments, Class<?>[] methodArgumentTypes, Clause[] methodTestSentence, String incorrectMethodStructureErrorMessage) {
+        this.methodClass = methodClass;
+        this.methodName = methodName;
+        this.arguments = arguments;
+        this.methodArgumentTypes = methodArgumentTypes;
+        this.methodTestSentence = methodTestSentence;
+        this.incorrectMethodStructureErrorMessage = incorrectMethodStructureErrorMessage;
+        setMethodOutput();
+    }
+
+    public MethodTest(Class<?> methodClass, String methodName, Clause[] methodTestSentence, String incorrectMethodStructureErrorMessage) {
+        this.methodClass = methodClass;
+        this.methodName = methodName;
+        this.arguments = null;
+        this.methodArgumentTypes = null;
+        this.methodTestSentence = methodTestSentence;
+        this.incorrectMethodStructureErrorMessage = incorrectMethodStructureErrorMessage;
+        setMethodOutput();
+    }
+
+    public Object invokeIfMethodExists() throws Throwable {
```

Comment:
I feel like we could just call this `callMethod()` or something so it makes more sense from a test write perspective. The term "invoke" only makes sense because I know that's what's under the hood but ideally test writers aren't looking at that.

You could implement this by leaving this method as is and just adding another method:
```java
public Object callMethod() throws Throwable {
    return invokeIfMethodExists();
}
```

Reply:
I can also just change to rename the Method now because we are using a new file now to do the testing entirely.